taf/repository_tool.py

```diff
@@ -173,31 +261,157 @@ def set_metadata_expiration_date(self, role, start_date=datetime.datetime.now(),
     expiration_date = start_date + datetime.timedelta(interval)
     role_obj.expiration = expiration_date
 
-  def write_roles_metadata(self, role, keystore, update_snapshot_and_timestamp=False):
+  def update_snapshot_and_timestmap(self, keystore, **kwargs):
+    """Update snapshot and timestamp metadata.
+
+    Args:
+      - keystore(str): Path to keystore with snapshot and timestamp keys
+      - kwargs(dict): (Optional) Expiration dates and intervals
+
+    Returns:
+      None
+
+    Raises:
+      - InvalidKeyError: If wrong key is used to sign metadata
+      - MetadataUpdateError: If any other error happened during metadata update
     """
-    Generates metadata of just one metadata file, corresponding
-    to the provided role
-    args:
-      repository: a tu repository
-      role: a tuf role
-      keystore: location of the keystore file
-      update_snapshot_and_timestamp: should timestamp and snapshot.json also be updated
+    try:
+      snapshot_date = kwargs.get('snapshot_date', datetime.datetime.now())
+      snapshot_interval = kwargs.get('snapshot_interval', None)
+
+      timestamp_date = kwargs.get('timestamp_date', datetime.datetime.now())
+      timestamp_interval = kwargs.get('timestamp_interval', None)
+
+      snapshot_key = load_role_key(keystore, 'snapshot')
+      self.update_snapshot(snapshot_key, snapshot_date, snapshot_interval, write=False)
+
+      timestamp_key = load_role_key(keystore, 'timestamp')
+      self.update_timestamp(timestamp_key, timestamp_date, timestamp_interval, write=False)
+
+      self._repository.writeall()
```

Comment:
`writeall` updates all dirty roles. You can take a look at `test_repository_tool.py` first test, `targets` data remain the same.

We have following metadata updates for now:

1. timestamp (periodically; should call `write`)
2. snapshot + timestamp
3. targets

Since targets metadata could be signed just with YubiKey, `writeall` will raise exception if targets metadata is dirty and valid YubiKey is not inserted.

I can change API to call write just for updated metadata and create a new method `writeall` which just calls TUF's repository `writeall` method. Then we can call it manually if needed. Does that make sense?

Reply:
I think that the three types of metadata updates are fine. I am just not sure about calling `writeall` in `update_snapshot_and_timestmap`, which means that targets will also be written here. I think that can be an unexpected effect of this function. It makes more sense to me to call `writeall` from `update_targets`, since that's closer to tuf's flow (when targets are updated snapshot and timestamp should be updated, not the other way around)

Reply:
`writeall` won't write unchanged metadata files, additionally it will raise an error if targets metadata file is dirty, and the targets signing yubikey is not present. The question is do we want to raise that error?

Reply:
Changed. `writeall` should be called explicitly if needed.